server/resolvers/moduleResolvers.ts

```diff
@@ -11,9 +11,15 @@ export class moduleResolvers {
         @Arg("house_key") house_key: string,
         @Arg("modules") modules: string,
     ){
-        let houseToBeUpdated = await Houses.findOne({ where: { house_key }})
-        houseToBeUpdated.modules = modules
+        let modulesToBeSaved : any = ImportedModules
+        let parsedModules = JSON.parse(modules)
 
+        for (const eachModule in parsedModules) {
+            modulesToBeSaved[eachModule].purchased = parsedModules[eachModule].purchased
+        }
+   
+        let houseToBeUpdated = await Houses.findOne({ where: { house_key }})
+        houseToBeUpdated.modules = JSON.stringify(modulesToBeSaved)
```

Comment:
We determine whether to show the app on the nav bar or not from the `object.purchased`

So it is now grabbing the new modules.json, and updating the purchased with the one currently in the app. This should add any new modules from the JSON file, while keeping the purchased settings of the user.

Reply:
Two things: 

- 'eachModule' should just be 'module'
- What is happening on line 14 with the colon and the 'any'

Once I get a better grasp on what's going on here, I'll approve the ones that git rid of the "update modules (admin)" workaround below!

Reply:
I meant to say "get rid of..." but "git rid of" is an alltime great typo

Reply:
I think keeping it as eachModule is fine. It's only used in the for loop afterall. plus, module I think is too similar to modules payload. I'm just trying to keep the variable names clear. 

The line 14 colon any is some typescript thing I'm nnot really sure my self. It would not let me do line 18 without it...

Reply:
OK yep I see what's happening and I kind of understand the typescript thing too. No problems there.

Let's check back in on naming conventions in a second, but what is the type of parsedModules?

Reply:
That's a good question, it didn't throw an error when I did it so I didn't add anything. I can try to add the type later just for the sake of consistency

Reply:
So I think it should be an array, yeah? In which case you should be using a for...of loop?

Reply:
that wouldn't work. I need the key not the content of the key. And it is still an object so I would need the for in loop. (I tried both ways, I actually forgot for of and for in loops already lol)

About the type of it, I can play around with it more, but that is very typescript thing and I'm not really good with it yet.

Reply:
In term of naming conventions. I will argue that since the app is supposed to be scalable, we will need lots of variable names. So I highly recommend longer more specific names rather than shorter and more vague ones. 

I would even say we should change eachModule to eachModuleKey to be even more specific.

Reply:
OK so yeah if it's an object not an array you need for...in. Understood. And don't worry about playing around with the typing, I was wrong about how parsedModules looks after you JSON parse it.

So last thing then is the naming convention. I think we need to follow best practice which is that in a for...in (or for...of) loop, the variable name should be a noun describing the iteration. check the mdn docs to see examples. 

This sort of thing will be looked at by prospective employers for sure, so we should really strive for best practice whenever we can!

Reply:
Wouldn't eachModuleKey be a noun describing the iteration? if you think it's too long then how about just 'key' then?

Reply:
eachModuleKey is a formulation, a sentance fragment. moduleKey would be fine, but I don't think that's actually what's being returned? Like, if you console logged eachModule right now, would you get a whole object containing the various fields contained in each object? If that's the case, we should stick with module or moduleObj. I might be reading it wrong though, what exactly does an instance of eachModule look like?

Within the declaration of a for...of loop the naming best practice is to show us exactly what it is thats being returned each time, so the fact that the variable might be in use elsewhere isn't a problem. It's scoped strictly to the loop, and a reader expects to find what it is that's returning with each iteration.

Reply:
if u console log it its just the key not the object. 'chat', 'direct messages', etc.

Reply:
And I like "moduleKey" that's much better

Reply:
ok cool I'm having trouble visualizing what that looks like but I'll play with it when it's merged. I think if you switch the variable name I'm good to go!